### ISSUE TITLE
Magento2.8 Admin panel add product is not working when we change attribute set

### ISSUE THREAD

**Issue by durgagupta** [magento/magento2#10705]
I have created a couple of attribute sets and now in Add product section when I switch the product attribute set the loader keeps on loading. when i see in console i see following error.

Uncaught TypeError: Cannot read property 'length' of undefined
on file : http://local.magento218/pub/static/version1504078942/adminhtml/Magento/backend/en_US/Magento_Ui/js/form/element/multiselect.js

at line no 52
which is 

        setPrepareToSendData: function (data) {
            if (!data.length) { // line of error
                data = '';
            }
            this.source.set(this.dataScope + '-prepared-for-send', data);
        },

### Preconditions
2. tested browser chrome 60.0.311, Firefox, also tested in Firefox private mode

### Steps to reproduce
In default sample data.
1. go to Product >> catalog >> Add product 
2. select Attribute set "Bag" then again change attribute set to "Bottom" then loader keeps on loading and loading.

Updated information:
Preconditions: Magento version: 2.2 and older version

Steps to reproduce:
Create an attribute with a multiselect and do not add any options to select.
It is not needed to assign it to an attribute set, even if it is not assigned, it will give the error.

Expected result
Possibility to add new products with another Attribute set then default.

 Actual result
Only the default product set will work now, in all other cases the 'Please wait' will stay in screen because of a JS error mentioned above.

**Comment by jlyshoel** [comment]
We've also encountered this problem in one of our stores. I've created a workaround by changing the line 
from ```if (!data.length) {``` to ```if (!data || !data.length) {```

```
setPrepareToSendData: function (data) {
    if (!data || !data.length) {
        data = '';
    }

    this.source.set(this.dataScope + '-prepared-for-send', data);
},
```

Then simply placed and deployed it in an extensions (attached), overwriting the original file. 
[workaround.zip](https://github.com/magento/magento2/files/1287721/workaround.zip)

**Comment by magento-engcom-team** [comment]
@durgagupta thank you for your report.
Please use the issue [reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) to report an issue. Please, edit issue description and include the preconditions, the steps required to reproduce, the actual result, and the expected result.
Please, add the following information:
 - [x] Preconditions: Magento version
 - [x] Steps to reproduce
 - [x] Expected result
 - [x] Actual result

**Comment by Tantuss** [comment]
Preconditions: Magento version: 2.2 and older version

Steps to reproduce:
Create an attribute with a multiselect and do not add any options to select.
It is not needed to assign it to an attribute set, even if it is not assigned, it will give the error.

Expected result
Possibility to add new products with another Attribute set then default.

 Actual result
Only the default product set will work now, in all other cases the 'Please wait' will stay in screen because of a JS error mentioned above.

**Comment by magento-engcom-team** [comment]
@durgagupta, thank you for your report.
The issue is already fixed in develop branch
But we will consider to backport the fix to patch releases

**Comment by imrandiligent** [comment]
Hi,
I am also getting the same error, could you please tell me the steps to resolve the issue

**Comment by Tantuss** [comment]
Remove all attributes which have a dropdown, but do not have a value to select.
Or add options to select.

**Comment by Mobecls** [comment]
#mageconf

**Comment by magento-engcom-team** [comment]
@Mobecls thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by Mobecls** [comment]
I've tried next steps:
1. Clone 2.2-develop branch.
2. Install sampledata.
3. Set developer mode.
4. Create Attribute Set named Test.
5. Create dropdown, multiselect attributes without options. Assigned them to custom Attribute set Test.
6. Tried to change options of attribute Tax Class and changed Attributes sets from Test to Bag and from Bag to Test.
7. Tested in production mode.

There is no such error in console.
